src/components/NavBar/NavRoutes.jsx

```diff
@@ -0,0 +1,34 @@
+import React from "react";
+import { Route, Redirect } from "react-router-dom";
+
+import Create from "../../pages/Studio.jsx";
+import Home from "../../pages/Home.jsx";
+import Playlist from "../../pages/Playlist.jsx";
+import Profile from "../../pages/Profile.jsx";
+import Register from "../../pages/Register.jsx";
+import Login from "../../pages/Login.jsx";
+import Logout from "../Logout.jsx";
+import { useAuthentication } from "../../contexts/AuthenticationContext.jsx";
+
+function NavRoutes() {
+  const routeTo = (path, componentDisplayed) => (
+    <Route exact path={path} component={componentDisplayed} />
+  );
+
+  const { authentication } = useAuthentication;
+  return (
+    <main>
+      {routeTo("/", Home)}
+      {routeTo("/profile", Profile)}
+      {routeTo("/create", Create)}
+      <Route exact path="/library">
+        {authentication ? { Playlist } : <Redirect to="/login" />}
+      </Route>
```

Comment:
if you must do something like that, declare a component before NavRoutes:

    function PlaylistIfLogged() {
      const { authentication } = useAuthentication();
      return authentication ? <Playlist /> : <Redirect to="/login" />;
    }

then you can use it:

    {routeTo("/library", PlaylistIfLogged)}